### PR TITLE
Support response model composition 

### PR DESCRIPTION
**Pull request by whunmr**
**Describe the PR**
Add support for inheritance/Model Composition in Response with allOf and ref.

syntax:
```go
// @success 200 {object} jsonresult.JSONResult{data=proto.Order} "desc"
// @success 200 {object} jsonresult.JSONResult{data=[]proto.Order} "desc"
// @success 200 {object} jsonresult.JSONResult{data=string} "desc"
// @success 200 {object} jsonresult.JSONResult{data=[]string} "desc"
```

please see generated result in testcases of this PR.

**Relation issue**
https://github.com/swaggo/swag/issues/650

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -500,6 +500,32 @@ type Account struct {
     Name string `json:"name" example:"account name"`
 }
 ```
+
+### Model composition in response
```
Comment (sdghchj) [inline README.md]: Also need to add a hyperlink in Content.Examples at the head of README.md

Comment (whunmr) [reply README.md]: sure, linked added.

**Comment by sdghchj** [comment]
Thank you for your contribution.

File: operation.go [hunk 2]
```diff
@@ -630,7 +630,71 @@ func findTypeDef(importPath, typeName string) (*ast.TypeSpec, error) {
...
+
+func (nested *nestedField) getSchema() *spec.Schema {
+	if IsGolangPrimitiveType(nested.Type) {
+		return &spec.Schema{SchemaProps: spec.SchemaProps{Type: []string{nested.Type}}}
+	}
+
+	return &spec.Schema{SchemaProps: spec.SchemaProps{Ref: nested.Ref}}
+}
+
+func (nested *nestedField) fillNestedSchema(response spec.Response, ref spec.Ref) {
```
[the remark this answers is not in the document]
Comment (whunmr) [reply operation.go]: Good idea,  I saw you have already refactored the code.   Thanks

File: operation.go [hunk 3]
```diff
@@ -630,7 +630,71 @@ func findTypeDef(importPath, typeName string) (*ast.TypeSpec, error) {
...
+	} else {
+		props[nested.Name] = *nested.getSchema()
+	}
+	nestedSpec := spec.Schema{
+		SchemaProps: spec.SchemaProps{
+			Type:       []string{"object"},
+			Properties: props,
+		},
+	}
+	response.Schema.AllOf = make([]spec.Schema, 0)
```
[the remark this answers is not in the document]
Comment (whunmr) [reply operation.go]: Thanks for the suggestions.

File: operation.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (sdghchj) [reply operation.go]: 昨晚就review了，看你没动，我自己帮改了，强迫症

**Comment by wujie1993** [comment]
awsome!!!

**Comment by sdghchj** [comment]
There is a bug for int, int32, int64 etc. which should be replaced with integer. 
@whunmr Hope you fix it, please.
```
// @success 200 {object} jsonresult.JSONResult{data=int} "desc"
// @success 200 {object} jsonresult.JSONResult{data=[]int} "desc"
```

doc for data=int:
```
"responses": {
        "200": {
            "description": "Error message, if code != 200",
            "schema": {
                "allOf": [
                    {
                        "$ref": "#/definitions/model.CommonHeader"
                    },
                    {
                        "type": "object",
                        "properties": {
                            "data": {
                                "type": "int"   //here should be integer
                            }
                        }
                    }
                ]
            }
        }
    }
```

**Pull request merged**